File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResults;
import entity.Result;

@RestController
@RequestMapping("/brand")
public class BrandController {
	
	@Reference
	private BrandService brandService;
	
	/**
	 * 查询所有品牌
	 */
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		return brandService.findAll();
	}
	
	/**
	 * 分页查询所有品牌
	 */
	@RequestMapping("/findPaging")
	public PageResults findPaging(int pageNum, int pageSize){
		return brandService.findPaging(pageNum,pageSize);
	}
	
	/**
	 * 添加品牌
	 * @param tbBrand
	 * @return
	 */
	@RequestMapping("/addTbBrand")
	public Result findPaging(@RequestBody TbBrand tbBrand){
		try {
			brandService.addTbBrand(tbBrand);
			return new Result(true, "添加成功");
		} catch (Exception e) {
			// TODO: handle exception
			return new Result(false, "添加失败");
		}
	}
	
	/**
	 * 根据id查询品牌
	 */
	@RequestMapping("/findOneTbBrandById")
	public TbBrand findOneTbBrandById(long tbBrandId){
		return brandService.findOneTbBrandById(tbBrandId);
	}
	
	/**
	 * 修改品牌
	 */
	@RequestMapping("/updateTbBrand")
	public Result updateTbBrand(@RequestBody TbBrand tbBrand){
		try {
			brandService.updateTbBrand(tbBrand);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			// TODO: handle exception
			return new Result(false, "修改失败");
		}
	}
	
	/**
	 * 删除品牌
	 */
	@RequestMapping("/deleteTbBrandByIds")
	public Result deleteTbBrandByIds(long[] ids) {
		// TODO Auto-generated method stub
		try {
			brandService.deleteTbBrandByIds(ids);
			return new Result(true, "删除成功");
		} catch (Exception e) {
			// TODO: handle exception
			return new Result(false, "删除失败");
		}	
	}
	
	/**
	 * 分页条件查询所有品牌
	 */
	@RequestMapping("/searchPaging")
	public PageResults findPaging(@RequestBody TbBrand tbBrand,int pageNum, int pageSize){
		return brandService.findPaging(tbBrand,pageNum,pageSize);
	}
	
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList(){
		return brandService.selectOptionList();
	}
	

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("brandController",function($scope,$http,$controller,brandService){
	
	$controller("baseController",{$scope:$scope});
			
	$scope.findAll=function(){
		brandService.findAll().success(function(data) {
			$scope.list=data;
		});
	}
	
	
	//删除所选品牌
	$scope.deleteSelection=function(){
		if($scope.selectIds.length==0){
			alert("请选择需要删除的品牌");
			return;
		}
		if(confirm("是否确定删除所选品牌")){
			brandService.deleteSelection($scope.selectIds).success(function(data) {
				if(data.success){
					$scope.reloadList();
				}else{
					alert(data.message);
				}
				
			});
		}
	}
	
	/*新增品牌*/
	$scope.saveTbBrand=function(){
		var object=null;
		if($scope.entity.id!=null){
			object=brandService.updateTbBrand($scope.entity);
		}else{
			object=brandService.addTbBrand($scope.entity);
		}
		object.success(function(data) {
			if(data.success){
				alert(data.message);
				$scope.reloadList();
			}else{
				alert(data.message);
			}
		});
	}
	
	/*根据id查询品牌*/
	$scope.findOneTbBrandById=function(tbBrandId){
		brandService.findOneTbBrandById(tbBrandId).success(function(data) {
			$scope.entity=data
		});
	}
	
	
	
	/*分页查询所有品牌*/
	$scope.findPaging=function(pageNum,pageSize){
		brandService.findPaging(pageNum,pageSize).success(function(data) {
			$scope.list=data.rows;
			$scope.paginationConf.totalItems=data.total;
		});
	}
	
	/*初始化查询条件*/
	$scope.searchEntity={};
	
	
	/*分页条件查询所有品牌*/
	$scope.search=function(pageNum,pageSize){
		brandService.search(pageNum,pageSize,$scope.searchEntity).success(function(data) {
			$scope.list=data.rows;
			$scope.paginationConf.totalItems=data.total;
		});
	}
});